plugins/woocommerce/tests/api-core-tests/tests/customers/customers-crud.test.js

```diff
@@ -0,0 +1,451 @@
+const {
+	test,
+	expect
+} = require('@playwright/test');
+const {
+	customer
+} = require('../../data');
+
+/**
+ * Tests for the WooCommerce Customers API.
+ *
+ * @group api
+ * @group customers
+ *
+ */
+test.describe('Customers API tests: CRUD', () => {
+	let customerId;
+
+	test.describe('Retrieve after env setup', () => {
+
+		// when the environment is created, we have an admin user and 
+		// an subscriber user that can both be accessed through their ids
+		// admin user will have id 1 and subscriber user will have id 2
+		// neither of these are returned as part of the get all customers call
+		// unless the role 'all' is passed as a search param 
+        // but can be accessed by specific id reference
+		test('can retrieve admin user', async ({
+			request
+		}) => {
+			// call API to retrieve the previously saved customer
+			const response = await request.get('/wp-json/wc/v3/customers/1');
+			const responseJSON = await response.json();
+			expect(response.status()).toEqual(200);
+			expect(responseJSON.is_paying_customer).toEqual(false);
+			expect(responseJSON.role).toEqual('administrator');
+			expect(responseJSON.username).toEqual('admin');
+		});
+
+		test('can retrieve subscriber user', async ({
+			request
+		}) => {
+			// call API to retrieve the previously saved customer
+			const response = await request.get('/wp-json/wc/v3/customers/2');
+			const responseJSON = await response.json();
+			expect(response.status()).toEqual(200);
+			expect(responseJSON.is_paying_customer).toEqual(false);
+			expect(responseJSON.role).toEqual('subscriber');
+		});
+
+		test('retrieve user with id 0 is invalid', async ({
+			request
+		}) => {
+			// call API to retrieve the previously saved customer
+			const response = await request.get('/wp-json/wc/v3/customers/0');
+			const responseJSON = await response.json();
+			expect(response.status()).toEqual(404);
+			expect(responseJSON.code).toEqual('woocommerce_rest_invalid_id');
+			expect(responseJSON.message).toEqual('Invalid resource ID.');
+		});
+
+		test('can retrieve customers', async ({
+			request
+		}) => {
+			// call API to retrieve all customers should initially return empty array
+			const response = await request.get('/wp-json/wc/v3/customers');
+			const responseJSON = await response.json();
+			expect(response.status()).toEqual(200);
+			expect(Array.isArray(responseJSON));
+			expect(responseJSON.length).toEqual(0);
+		});
+
+		// however, if we pass in the search string for role 'all' then all users are returned
+		test('can retrieve all customers', async ({
+			request
+		}) => {
+			// call API to retrieve all customers should initially return empty array
+            // unless the role 'all' is passed as a search string, in which case the admin
```

Comment:
Minor nit on the spacing on the comment lines, here and on 78 below. Just need to tab over so they match on 76 (looks like it may be a spaces vs tab case).

Reply:
Good shout - reformatted the page - hopefully resolves this spacing issue!

Reply:
Looks good to me!